File: cpp/algo/greedy_11399.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
  /// Input
  int n;
  vector<int> input(1000);
  cin >> n;
  for (int i = 0; i < n; ++i) {
    cin >> input[i];
  }

  /// Sort
  sort(input.begin(), input.begin() + n);

  /// Sum
  int sum = input[0];
  int prev = input[0];

  for (int i = 1; i < n; ++i) {
    prev = prev + input[i];
    sum += prev;
  }
  cout << sum << endl;
  return 0;
}

File: python/algo/baekjoon/greedy/greedy_11399.py
# https://www.acmicpc.net/problem/11399

# 5
# 3 1 4 3 2

# 32

if __name__ == '__main__':
    N = int(input())
    l = list(map(int, input().split()))
    l.sort()
    ans = l[0]
    for i in range(1, len(l)):
        # 누적= 누적 + 앞구간의 시간 + 현재 걸린 시간
        ans = ans + sum(l[0:i]) + l[i]

    print(ans)
    # // 맨앞에껀 기다리는 시간 0
    #  cnt = cnt + 신규 시간